### PR TITLE
Add grafana agent k8s to observability plugin 

### PR DESCRIPTION
**Pull request by chanchiwai-ray**
This PR add grafana-agent-k8s to the observability plugin.

- Add grafana-agent-k8s application
- Add relations to COS
  - grafana-agent-k8s:grafana-dashboards-provider                   grafana-dashboards:grafana-dashboard
  - loki-logging:logging                                                                 grafana-agent-k8s:logging-consumer
  - prometheus-receive-remote-write:receive-remote-write         grafana-agent-k8s:send-remote-write

### CODE REVIEW

File: sunbeam-python/sunbeam/plugins/observability/etc/deploy-cos/main.tf [hunk 1]
```diff
@@ -129,7 +129,7 @@ resource "juju_application" "loki" {
 
 # juju integrate traefik prometheus
 resource "juju_integration" "traefik-to-prometheus" {
-  model    = var.model
+  model = var.model
```
Comment (samuelallan72) [inline sunbeam-python/sunbeam/plugins/observability/etc/deploy-cos/main.tf]: This should probably be left for a future refactoring/tidying commit, to have a smaller diff and make it clearer what the functional changes are?

File: sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent-k8s/main.tf [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+terraform {
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = "= 0.8.0"
```
Comment (samuelallan72) [inline sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent-k8s/main.tf]: this version is quite old.  Might be worth asking the sunbeam team if it's ok to use the latest version.

File: sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent-k8s/main.tf [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  config  = var.cos-state-config
+}
+
+resource "juju_application" "grafana-agent-k8s" {
+  name  = "grafana-agent-k8s"
+  model = var.model
+
+  charm {
+    name     = "grafana-agent-k8s"
+    series     = "jammy" # we may need to use "base" in the future
```
Comment (samuelallan72) [inline sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent-k8s/main.tf]: `series` is still available (though deprecated) in the latest release.  So we can make this comment more precise with something like:

```suggestion
    series   = "jammy"
    # after upgrading to juju provider >= 0.10.0, replace series with:
    # base = "ubuntu@22.04"
```

**Review by samuelallan72 — APPROVED** [review]
Code changes look good to me.  I left a few nonblocking comments and suggestions.  I'm still in process of building and testing the snap myself though.

File: sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent-k8s/main.tf (hunk 2 above)
Comment (agileshaw) [reply sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent-k8s/main.tf]: +1. Not sure what is Sunbeam's plan for updating the providers across all terraform configurations, but we should at least file an issue if they don't want to use a newer provider version

File: sunbeam-python/sunbeam/plugins/observability/etc/deploy-cos/main.tf (hunk 1 above)
Comment (chanchiwai-ray) [reply sunbeam-python/sunbeam/plugins/observability/etc/deploy-cos/main.tf]: Agree, I don't really want them to think it's a bit change

File: sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent-k8s/main.tf (hunk 2 above)
Comment (chanchiwai-ray) [reply sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent-k8s/main.tf]: They might do a complete change from the snap perspective. So I will stick with it now. But asking does not hurt 😃

Comment (chanchiwai-ray) [reply sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent-k8s/main.tf]: well, it's not a "might". They are doing it...

**Review by agileshaw — APPROVED** [review]
Internally approved and we are waiting for the PR against to be merged

**Pull request merged**